src/charm.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python3
+
+import logging
+
+from ops.main import main
+from ops.pebble import Layer
+from ops.charm import CharmBase
+from ops.model import ActiveStatus, WaitingStatus, MaintenanceStatus
+
+logger = logging.getLogger(__name__)
+
+
+class TrainingOperatorCharm(CharmBase):
+    """A Juju Charm for Training Operator"""
+
+    def __init__(self, *args):
+        super().__init__(*args)
+
+        self.logger = logging.getLogger(__name__)
+
+        self._name = self.model.app.name
+        self._namespace = self.model.name
```

Comment:
For the sake of simplicity I prefer the charm to assume it will never be deployed in the `controller` model. The documentation and the juju cli tool clearly state the user should create a new namespace and there is not a reason this charm should ever be in the controller namespace. 

Therefore I prefer to keep assuming the namespace will be the model name and keep the code simpler.

Reply:
Sure :)